drivers/serial/Kconfig.nrfx

```diff
@@ -72,9 +87,88 @@ config UART_0_NRF_RTS_PIN
 	help
 	  The GPIO pin to use for RTS.
 
-endif # UART_0_NRF_UART
+endif # UART_0_NRF_UART || UART_0_NRF_UARTE
+
+# ----------------- port 1 -----------------
+config UART_1_NRF_UARTE
+	bool "nRF UARTE 1"
+	depends on HAS_HW_NRF_UARTE1
+	help
+	  Enable nRF UART with EasyDMA on port 1.
+
+if UART_1_NRF_UARTE
+
+config UART_1_INTERRUPT_DRIVEN
+	bool "Enable UART Interrupt support on port 1"
+	depends on UART_INTERRUPT_DRIVEN
+	default y
+	help
+	  This option enables UART interrupt support on port 1.
+
+config UART_1_NRF_PARITY_BIT
+	bool "Enable parity bit"
+	help
+	  Enable parity bit.
+
+config UART_1_NRF_FLOW_CONTROL
+	bool "Enable flow control"
+	help
+	  Enable flow control. If selected, additionally two pins, RTS and CTS
+	  have to be configured.
+
+config UART_1_NRF_TX_PIN
+	int "TX Pin Number"
+	range 0 47 if SOC_NRF52840_QIAA
+	range 0 31
+	help
+	  The GPIO pin to use for TX.
+
+config UART_1_NRF_RX_PIN
+	int "RX Pin Number"
+	range 0 47 if SOC_NRF52840_QIAA
+	range 0 31
+	help
+	  The GPIO pin to use for RX.
+
+config UART_1_NRF_CTS_PIN
+	int "CTS Pin Number"
+	range 0 47 if SOC_NRF52840_QIAA
+	range 0 31
+	depends on UART_1_NRF_FLOW_CONTROL
+	help
+	  The GPIO pin to use for CTS.
+
+config UART_1_NRF_RTS_PIN
+	int "RTS Pin Number"
+	range 0 47 if SOC_NRF52840_QIAA
+	range 0 31
+	depends on UART_1_NRF_FLOW_CONTROL
+	help
+	  The GPIO pin to use for RTS.
+
+endif # UART_1_NRF_UARTE
+
+if NRF_UARTE_PERIPHERAL
+
+config UART_NRFX_TX_BUFFER_SIZE
```

Comment:
No, buffer size value is up to the user and it equals `UART_NRFX_TX_BUFFER_SIZE`.

If user will configure `UART_NRFX_TX_BUFFER_SIZE` higher than allowed tx buffer size by architecture, driver will limit it to max allowed by architecture value.

Maximum buffer size is limited by seleceted SoC and its value is provided by Nordic MDK. In fact this can be kept in DTS but not `UART_NRFX_TX_BUFFER_SIZE`.

Reply:
Why isn't this buffer per UART instance?

Reply:
I will update that.

Reply:
fixed